### PR TITLE
Adding Pascal raw dataset

### PR DESCRIPTION
**Pull request by hossein-dl**
Adding pascal raw dataset. 

-> **Depends on https://github.com/Deeplite/deeplite-torch-zoo/pull/342/files to finalize (rawpy module needs to be added)**

### CODE REVIEW

**Comment by lzrwch** [comment]
@hossein-dl you can add rawpy to setup.py here as well

File: tests/test_raw_dataset_loading.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import shutil
+from pathlib import Path
+
+import pytest
+
+from deeplite_torch_zoo import get_dataloaders
+
+
+DATASETS_ROOT = Path('/public-shared/datasets/RAW_DATASETS/custom_data/pascal_cropped_clf/raw')
```
Comment (lzrwch) [inline tests/test_raw_dataset_loading.py:9]: we have no `/public-shared/` in CI. please create a small downloadable version of this dataset for testing, upload to our S3, and have it downloaded automatically in order to run tests in CI

File: deeplite_torch_zoo/api/datasets/classification/pascalraw.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        batch_size=test_batch_size,
+        is_training=False,
+        num_workers=num_workers,
+        distributed=distributed,
+        img_dtype=torch.float16,
+    )
+
+    return {"train": train_loader, "test": test_loader}
+
+class PASCALRAWDataset(Dataset):
```
Comment (lzrwch) [inline deeplite_torch_zoo/api/datasets/classification/pascalraw.py:60]: please add the dataset class definition [here](https://github.com/Deeplite/deeplite-torch-zoo/tree/develop/deeplite_torch_zoo/src/classification/datasets). if the code was taken from a third-party source, place it in a separate subdirectory there with a corresponding LICENSE file

File: deeplite_torch_zoo/api/datasets/classification/pascalraw.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    fp16=False,
+    distributed=False,
+    train_transforms=None,
+    val_transforms=None,
+    **kwargs,
+):
+    if data_root == "":
+        data_root = os.path.join(expanduser("~"), ".deeplite-torch-zoo")
+    image_folder = data_root
+    train_dataset = PASCALRAWDataset(root=image_folder, split='train', transform=train_transforms)
```
Comment (lzrwch) [inline deeplite_torch_zoo/api/datasets/classification/pascalraw.py:34]: transforms can only be passed by the user, no augmentation by default?

File: deeplite_torch_zoo/api/datasets/classification/pascalraw.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    test_dataset = PASCALRAWDataset(root=image_folder, split='test', transform=val_transforms)
+
+    train_loader = create_loader(
+        train_dataset,
+        input_size=[224, 224, 3],
```
Comment (lzrwch) [inline deeplite_torch_zoo/api/datasets/classification/pascalraw.py:39]: pls pass it as an arg with default value instead of hardcoding

**Review by lzrwch — CHANGES_REQUESTED** [review]

**Comment by lzrwch** [comment]
@hossein-dl why is this added to classification, is it a version of the dataset converted to a multi-label classification problem?

File: deeplite_torch_zoo/api/datasets/classification/pascalraw.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        raw_image = image.raw_image
+        raw_image = raw_image.astype(np.float16)
+
+        if self.transform:
+            raw_image = self.transform(raw_image)
+        else:
+            image = ToTensor()(raw_image)
+
+        label = Path(image_path).parent.stem
+        return image, label
```
Comment (lzrwch) [inline deeplite_torch_zoo/api/datasets/classification/pascalraw.py:97]: labels have to be read to from a file and converted to a corresponding format, I suppose

**Review by lzrwch — CHANGES_REQUESTED** [review]

**Comment by hossein-dl** [comment]
> @hossein-dl why is this added to classification, is it a version of the dataset converted to a multi-label classification problem?

Yes, @shahrukh-dl has created a classification dataset from the original PascalRaw dataset.

**Comment by lzrwch** [comment]
> > @hossein-dl why is this added to classification, is it a version of the dataset converted to a multi-label classification problem?
> 
> Yes, @shahrukh-dl has created a classification dataset from the original PascalRaw dataset.

Then please change the name to something like `PASCALRAWMultiLabelCls`

File: deeplite_torch_zoo/api/datasets/classification/pascalraw.py (hunk 5 above)
Comment (lzrwch) [reply deeplite_torch_zoo/api/datasets/classification/pascalraw.py:97]: keeping in mind that this is multi-label cls (e.g. each label would be a tuple of ints)

**Comment by alexh-deeplite** [comment]
@hossein-dl @lzrwch  I'm working on the object detection version of this dataset based on the YoloDataset class, and will submit as a separate PR when it is tested. Just wanted to let you know

**Comment by lzrwch** [comment]
@alexh-deeplite @hossein-dl closing as per today's discussion
